runtime/src/main/java/org/corfudb/runtime/collections/PersistedStreamingMap.java

```diff
@@ -0,0 +1,320 @@
+package org.corfudb.runtime.collections;
+
+import com.google.common.collect.Streams;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuError;
+import org.corfudb.util.serializer.ISerializer;
+import org.rocksdb.RocksDB;
+import org.rocksdb.Options;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.RocksIterator;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A concrete implementation of {@link StreamingMap} that is capable of storing data
+ * off-heap. The location for the off-heap data is provided by {@link File} dataPath,
+ * while the resource policy (memory and storage limits) are defined in {@link Options}.
+ *
+ * @param <K> key type
+ * @param <V> value type
+ */
+@Slf4j
+public class PersistedStreamingMap<K, V> implements StreamingMap<K, V> {
+    private static final List<PersistedStreamingMap> openedResources = new ArrayList<>();
+
+    private final AtomicInteger dataSetSize = new AtomicInteger();
+    private final CorfuRuntime corfuRuntime;
+    private final ISerializer serializer;
+    private final RocksDB rocksDb;
+
+    public PersistedStreamingMap(@NonNull Path dataPath,
+                                 @NonNull Options options,
+                                 @NonNull ISerializer serializer,
+                                 @NonNull CorfuRuntime corfuRuntime) {
+        try {
+            RocksDB.destroyDB(dataPath.toFile().getAbsolutePath(), options);
```

Comment:
destroyingDb in a constructor - it seems like not the best idea.
I would prefer just to check that if the DB is already exists then throw an exception.

PersistedStreamingMap sitting on a different level, it just can't create/delete/manage database, 
it can't know what's going on. It can only check that the enviroment is in correct state or not

Reply:
The client does not understand the how the persistence is achieved. Asking clients to cleanup the resources after a hard-crash would constitute a leaky design.